Sage-Bionetworks/schematic
# Fixed column headers problem when generating an Excel spreadsheet for getting an existing manifest

This is related to the issue #1148 

It turned out that this issue occurred because at one point we switched to a newer version of pandas, and our old syntax of populating an existing manifest using `pd.ExcelWriter ` no longer works. To modify an existing excel spreadsheet, we have to set `if_sheet_exists` parameter. 

Also, I noticed that in our old logic, if we generate an Excel spreadsheet. we have something like: 
```
        # Populate empty template with existing manifest
        if manifest_record:

            # TODO: Update or remove the warning in self.__init__() if
            # you change the behavior here based on self.use_annotations
                            
            # populate empty manifest with content from downloaded/existing manifest
            manifest_sh = self.set_dataframe_by_url(empty_manifest_url, manifest_record[1])

            # determine the format of manifest
            result = self._handle_output_format_logic(output_format = output_format, output_path = output_path, sheet_url = sheet_url, manifest_url = manifest_sh.url, empty_manifest_url=empty_manifest_url, dataframe = manifest_record[1])
            return result
```

This means that when a manifest record exists, we would first populate an empty manifest URL with content from an existing manifest before exporting the manifest to Excel. This shouldn't happen. To avoid sending metadata to Google, I changed the logic so that `set_dataframe_by_url` only gets used when the output format is set to Google sheet

To test, I did the following: 
* Tested it out manually on swagger UI: 
   - example data model
   - existing dataset id: syn51200926
   - data type: Patient
   - output format: excel
   - asset view: syn51200927

* Using CLI to get an existing manifest: `schematic manifest --config ~/path/to/config.yml get --output_xlsx 'test.xlsx' --dataset_id syn51200926`
* Ran tests related to generating a new manifest and getting an existing manifest in test_api.py

To do: 
* Add tests in test_api.py that check if Excel spreadsheet has column headers.
